tests/cli-v2.rs

```diff
@@ -1374,3 +1374,53 @@ fn check_pgp_keys() {
         );
     })
 }
+
+#[test]
+fn install_allow_downgrade() {
+    clitools::setup(Scenario::MissingComponent, &|config| {
+        let trip = TargetTriple::from_build();
+
+        // this dist has no rls and there is no newer one
+        set_current_dist_date(config, "2019-09-14");
+        expect_ok(
+            config,
+            &[
+                "rustup",
+                "toolchain",
+                "install",
+                "nightly",
+                "--no-self-update",
+            ],
+        );
+        expect_err(
+            config,
+            &[
+                "rustup",
+                "toolchain",
+                "install",
+                "nightly",
+                "--no-self-update",
+                "-c",
+                "rls",
+            ],
+            &format!(
+                "component 'rls' for target '{}' is unavailable for download for channel nightly",
+                trip,
+            ),
+        );
+        expect_ok(
+            config,
+            &[
+                "rustup",
+                "toolchain",
+                "install",
+                "nightly",
+                "--no-self-update",
+                "-c",
+                "rls",
+                "--allow-downgrade",
+            ],
+        );
+        expect_stdout_ok(config, &["rustc", "--version"], "hash-nightly-2");
```

Comment:
While this is useful to check that it did downgrade, that'd be clearer if you'd done the equivalent above as commented.  Also you could usefully assert that the `rls` component is now available via `expect_component_executable()`

Reply:
I like the expect_component_executable check much more than the version check. Thx